controller/src/beerocks/master/db/db.cpp

```diff
@@ -2494,6 +2494,31 @@ bool db::get_bml_nw_map_update_enable(Socket *sd)
     return false;
 }
 
+bool db::set_bml_topology_update_enable(Socket *sd, bool update_enable)
+{
+    if (sd) {
+        for (auto it = bml_listeners_sockets.begin(); it < bml_listeners_sockets.end(); it++) {
+            if (sd == (*it).sd) {
+                (*it).topology_updates = update_enable;
+                return true;
+            }
+        }
+    }
+    return false;
+}
+
+bool db::get_bml_topology_update_enable(Socket *sd)
+{
+    if (sd) {
+        for (auto it = bml_listeners_sockets.begin(); it < bml_listeners_sockets.end(); it++) {
+            if (sd == (*it).sd) {
+                return (*it).topology_updates;
+            }
+        }
+    }
+    return false;
+}
```

Comment:
Please always use this style:
```cpp
if (error) {
    // handle error
}
// rest of the code
```

Also, if you choose to loop with iterators, please use `it->something` and not `(*it).something`.
But I would prefer that you loop with ranged loops:
```cpp
for (auto &soc : bml_listeners_sockets) {
    // Your code here
}
```